pkg/alert/alert.go

```diff
@@ -133,34 +134,31 @@ func NewQueue(logger log.Logger, reg prometheus.Registerer, capacity, maxBatchSi
 		toAddLset:       toAdd,
 		toExcludeLabels: toExclude,
 
-		dropped: prometheus.NewCounter(prometheus.CounterOpts{
+		dropped: promauto.With(reg).NewCounter(prometheus.CounterOpts{
 			Name: "thanos_alert_queue_alerts_dropped_total",
 			Help: "Total number of alerts that were dropped from the queue.",
 		}),
-		pushed: prometheus.NewCounter(prometheus.CounterOpts{
+		pushed: promauto.With(reg).NewCounter(prometheus.CounterOpts{
 			Name: "thanos_alert_queue_alerts_pushed_total",
 			Help: "Total number of alerts pushed to the queue.",
 		}),
-		popped: prometheus.NewCounter(prometheus.CounterOpts{
+		popped: promauto.With(reg).NewCounter(prometheus.CounterOpts{
 			Name: "thanos_alert_queue_alerts_popped_total",
 			Help: "Total number of alerts popped from the queue.",
 		}),
 	}
-	capMetric := prometheus.NewGaugeFunc(prometheus.GaugeOpts{
+	_ = promauto.With(reg).NewGaugeFunc(prometheus.GaugeOpts{
```

Comment:
Is it better to do `_ = func()` or just `func()`?

Reply:
Well I do `_` to indicate that this returns value, but we don't need it. Do you think it's wrong pattern?

Reply:
I like it as it avoid surprises (: But happy to change this in the following PRs :hugs:

Reply:
I just wanted to make sure we were being consistent in the project one way or the other so :+1: